### ISSUE TITLE
Faster loss functions based on #27

### ISSUE THREAD

**Issue by tejaswini** [IBM/pytorch-seq2seq#34]
Faster loss function similar to OpenNMT memory efficient loss. Instead of looping row by row evaluating the loss batch times. We transformed the target and output from 2D and 3D to 1D and 2D. Evaluated the loss once for the entire batch.
        # (seq len, batch size, dictionary size) -> (batch size * seq len, dictionary size)
        outputs = outputs.view(-1, outputs.size(2))
        # (seq len, batch size) -> (batch size * seq len)
        targets = targets.view(-1)
        self.criterion(outputs, targets)

**Comment by kylegao91** [comment]
@tejaswini Any progress on this?
